planner/core/find_best_task.go

```diff
@@ -1546,8 +1546,10 @@ func (is *PhysicalIndexScan) addPushedDownSelection(copTask *copTask, p *DataSou
 			}
 			tableSel.stats = copTask.plan().statsInfo().Scale(selectivity)
 		}
-		tableSel.SetChildren(copTask.tablePlan)
-		copTask.tablePlan = tableSel
+		if copTask.tablePlan != nil {
```

Comment:
`copTask.tablePlan` shouldn't be nil here since it must be an `IndexLookup`.
But we can add some defense code here: `if copTask.tablePlan == nil {return someErr}`.